### PR TITLE
Seed to trackster association, treatment of seeds with no trackster

### PR DESCRIPTION
**Pull request by felicepantaleo**
#### PR description:

This PR includes 

- a map for associating seeds to tracksters. 
- if a seed is not associated to any trackster, an empty trackster is created
- the p_T threshold for tracking iteration is lowered to 1GeV/c

#### PR validation:
tested on wf D49 `23289.0`, `23288.0` and ttbar PU200

### CODE REVIEW

**Comment by rovere** [comment]
Ciao @felicepantaleo,
do you have performance numbers, both from the CPU/Timing point of view and, also, from the physics point of view?
In particular, we should check that the timing is still under control even after lowering the p_T threshold and, also, check if the performance in terms of efficiency/fake/merge/duplicate did change after the move to a lower p_T. I'd naively expect to recover quite some efficiency, but it would be nice to have the proof of that.

File: RecoHGCal/TICL/plugins/PatternRecognitionbyCA.cc [hunk 1]
```diff
@@ -153,14 +160,16 @@ void PatternRecognitionbyCA::makeTracksters(const PatternRecognitionAlgoBase::In
...
-    mergeTrackstersTRK(result, input.layerClusters, tmp);
+    mergeTrackstersTRK(result, input.layerClusters, tmp, seedToTracksterAssociation);
     tmp.swap(result);
   }
 
   ticl::assignPCAtoTracksters(result, input.layerClusters, rhtools_.getPositionLayer(rhtools_.lastLayerEE()).z());
 
   // run energy regression and ID
   energyRegressionAndID(input.layerClusters, result);
+  emptyTrackstersFromSeedsTRK(result, seedToTracksterAssociation, input.regions[0].collectionID);
```
Comment (rovere) [inline RecoHGCal/TICL/plugins/PatternRecognitionbyCA.cc]: You mentioned at some point that you experience problems/crashes when passing these objects to ML for PID and energy regression. I assume that's why you put the call after `energyRegressionAndID`.
Maybe this should be documented in the code.
Also, the `TracksterMergeProducer` will run another loop of `energyRegressionAndID`: have you checked why, in that case, the code is not crashing?

**Comment by felicepantaleo** [comment]
some timing results on ttbar pu200:

| module                              | original  | PR        | difference (%) |
|-------------------------------------|-----------|-----------|----------------|
| ticlCandidateFromTrackstersProducer | 0.008759  | 0.011655  | 0.25           |
| ticlLayerTileProducer               | 0.045689  | 0.044659  | -0.02          |
| ticlSeedingGlobal                   | 2.5E-05   | 2.6E-05   | 0.04           |
| ticlSeedingTrk                      | 0.00338   | 0.004544  | 0.26           |
| trackstersEM                        | 1.009374  | 0.235263  | -3.29          |
| trackstersHAD                       | 1.538195  | 0.314291  | -3.89          |
| trackstersMIP                       | 5.68386   | 5.792387  | 0.02           |
| trackstersMerge                     | 1.070906  | 0.584841  | -0.83          |
| trackstersTrk                       | 1.117329  | 5.379495  | 0.79           |
| TOTAL                               | 10.477517 | 12.367161 | 0.15           |

**Comment by rovere** [comment]
> some timing results on ttbar pu200:
> 
> module	original	PR	difference (%)
> ticlCandidateFromTrackstersProducer	0.008759	0.011655	33.06
> ticlLayerTileProducer	0.045689	0.044659	-2.25
> ticlSeedingGlobal	2.5E-05	2.6E-05	4.00
> ticlSeedingTrk	0.00338	0.004544	34.44
> trackstersEM	1.009374	0.235263	-76.69
> trackstersHAD	1.538195	0.314291	-79.57
> trackstersMIP	5.68386	5.792387	1.91
> trackstersMerge	1.070906	0.584841	-45.39
> trackstersTrk	1.117329	5.379495	381.46
> TOTAL	10.477517	12.367161	18.04

It was somehow expected. The real question is if now the Trk iteration is slower but gathering real stuff (and stealing it from the other iterations) or it is just a more expensive garbage-collector.

**Comment by rovere** [comment]
@felicepantaleo after your rebase it's impossible to isolate your changes....

**Pull request closed without merging**